KarhooSDK/Api/Response/ServiceCancellationType.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  ServiceCancellationType.swift
+//  KarhooSDK
+//
+//  Created by Corneliu on 25.03.2021.
+//  Copyright © 2021 Flit Technologies Ltd. All rights reserved.
+//
+
+import Foundation
+
+public enum ServiceCancellationType: KarhooCodableModel {
+
+    case timeBeforePickup
+    case beforeDriverEnRoute
+    case other
```

Comment:
Is it possible to have enum(value: String) to preserve the actual value which is returned?

Reply:
In the API response, the cancellation type strings start with uppercase letters. When decoding String enums they are key sensitive.

The convention in Swift is that the case values should start with lowercase letters. I think it would be great to respect that standard convention.

The values are preserved because the model also implements the Encodidable protocol.

Reply:
```
public enum ServiceCancellationType: KarhooCodableModel {
    case timeBeforePickup
    case beforeDriverEnRoute
    case other(value: String)
    }
```
    
   ```
 public init(from decoder: Decoder) throws {
        let typeString = try decoder.singleValueContainer().decode(String.self)
        switch typeString {
        case "BeforeDriverEnRoute":
            self = .beforeDriverEnRoute
        case "TimeBeforePickup":
            self = .timeBeforePickup
        default:
            self = .other (value: typeString)
        }
    }
    
```

Reply:
Would this work? or something like this

Reply:
Ah, ok, I see what you mean. Yes, sure, it works like this.